test/e2e/placement_selecting_resources_test.go

```diff
@@ -592,3 +593,105 @@ var _ = Describe("validating CRP when deleting resources in a matching namespace
+				Name: crpName,
+				// Add a custom finalizer; this would allow us to better observe
+				// the behavior of the controllers.
+				Finalizers: []string{customDeletionBlockerFinalizer},
+			},
+			Spec: placementv1beta1.ClusterResourcePlacementSpec{
+				ResourceSelectors: []placementv1beta1.ClusterResourceSelector{
+					{
+						Group:   "",
+						Kind:    "Namespace",
+						Version: "v1",
+						Name:    "invalid-namespace",
+					},
+				},
+				Strategy: placementv1beta1.RolloutStrategy{
+					RollingUpdate: &placementv1beta1.RollingUpdateConfig{
+						UnavailablePeriodSeconds: pointer.Int(5),
+					},
+				},
+				RevisionHistoryLimit: pointer.Int32(1),
+			},
+		}
+		By(fmt.Sprintf("creating placement %s", crpName))
+		Expect(hubClient.Create(ctx, crp)).To(Succeed(), "Failed to create CRP %s", crpName)
+	})
+
+	AfterAll(func() {
+		By(fmt.Sprintf("deleting placement %s", crpName))
+		cleanupCRP(crpName)
+
+		By("deleting created work resources")
+		cleanupWorkResources()
+	})
+
+	It("should update CRP status as expected", func() {
+		crpStatusUpdatedActual := func() error {
+			return validateCRPStatus(types.NamespacedName{Name: crpName}, nil)
+		}
+		Eventually(crpStatusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update CRP %s status as expected", crpName)
+	})
+
+	It("adding resource selectors", func() {
+		updateFunc := func() error {
+			crp := &placementv1beta1.ClusterResourcePlacement{}
+			if err := hubClient.Get(ctx, types.NamespacedName{Name: crpName}, crp); err != nil {
+				return err
+			}
+
+			crp.Spec.ResourceSelectors = append(crp.Spec.ResourceSelectors, placementv1beta1.ClusterResourceSelector{
+				Group:   "",
+				Kind:    "Namespace",
+				Version: "v1",
+				Name:    fmt.Sprintf(workNamespaceNameTemplate, GinkgoParallelProcess()),
+			})
+			// may hit 409
+			return hubClient.Update(ctx, crp)
+		}
+		Eventually(updateFunc(), eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update the crp %s", crpName)
+	})
+
+	It("should update CRP status as expected", checkIfPlacedWorkResourcesOnAllMemberClusters)
+
+	It("should have one policy snapshot revision and one resource snapshot revision only", func() {
+		matchingLabels := client.MatchingLabels{placementv1beta1.CRPTrackingLabel: crpName}
+
+		snapshotList := &placementv1beta1.ClusterSchedulingPolicySnapshotList{}
+		Expect(hubClient.List(ctx, snapshotList, matchingLabels)).Should(Succeed(), "Failed to list the policy revisions")
+		Expect(len(snapshotList.Items)).Should(Equal(1), "clusterSchedulingPolicySnapshotList got %v, want 1", len(snapshotList.Items))
+
+		resourceSnapshotList := &placementv1beta1.ClusterResourceSnapshotList{}
+		Expect(hubClient.List(ctx, resourceSnapshotList, matchingLabels)).Should(Succeed(), "Failed to list the resource revisions")
+		Expect(len(snapshotList.Items)).Should(Equal(1), "clusterResourceSnapshotList got %v, want 1", len(snapshotList.Items))
+	})
+
```

Comment:
why not add the configMap and check that we will have 2 resource snapshot and 1 policy snapshot? Is this covered somewhere else?

Reply:
sure, will add one.